### PR TITLE
More cleanup

### PR DESCRIPTION
**Pull request by Eitot**
Removes most `#import` statements from *header* files (only in `src/`). Forward declaration (`@class`) is used for referenced types, as well as `#import` in the *implementation* files. This should make the dependencies clearer and avoid unnecessary imports.

Also replaces private class categories (`@interface X (Private)`) with extensions (`@interface X ()`). This has the advantage that `@property` declarations will be backed by an instance variable (named categories can't have storage), and that the compiler checks whether declared methods are actually implemented.

The instance variables in the Database class are replaced by private properties.

### CODE REVIEW

**Comment by Eitot** [comment]
6fc3392 causes a runtime crash when built with Xcode 8. Have to investigate.

**Comment by Eitot** [comment]
For the record: there was a problem with an asynchronous database call caused by the public getter `-(NSInteger)countOfUnread`. I sidestepped this problem by referencing the ivar (`_countOfUnread`) internally. Not ideal, may have another look at this in the future.

File: src/KeyChain.m [hunk 1]
```diff
@@ -19,8 +19,6 @@
 //
 
 #import "KeyChain.h"
-#import <Security/SecKeychain.h>
```
Comment (josh64x2) [inline src/KeyChain.m]: Are you sure we dont need to `@import Security;` for `SecKeychainFindInternetPassword` etc?

Comment (Eitot) [inline src/KeyChain.m]: Not sure, it seems to compile just fine. Probably doesn't hurt to at least link the framework.

**Pull request merged**